packages/shared/src/components/PostOptionsReadingHistoryMenu.tsx

```diff
@@ -0,0 +1,156 @@
+import React, { ReactElement, useContext } from 'react';
+import { Item } from '@dailydotdev/react-contexify';
+import dynamic from 'next/dynamic';
+import { QueryClient, QueryKey, useQueryClient } from 'react-query';
+import { ReadHistoryPost } from '../graphql/posts';
+import ShareIcon from '../../icons/share.svg';
+import BookmarkIcon from '../../icons/outline/bookmark_outlined.svg';
+import BookmarkIconFilled from '../../icons/filled/bookmark_filled.svg';
+import XIcon from '../../icons/outline/x.svg';
+import useBookmarkPost from '../hooks/useBookmarkPost';
+import AuthContext from '../contexts/AuthContext';
+import { ReadHistoryInfiniteData } from '../hooks/useInfiniteReadingHistory';
+import { useCopyLink } from '../hooks/useCopyLink';
+
+const PortalMenu = dynamic(() => import('./fields/PortalMenu'), {
+  ssr: false,
+});
+
+export type PostOptionsReadingHistoryMenuProps = {
+  post: ReadHistoryPost;
+  onHidden?: () => unknown;
+  displayCopiedMEssageFunc?: () => void;
+  onHide?: (postId: string) => unknown;
+};
+
+const MenuIcon = ({ Icon }) => {
+  return <Icon className="mr-2 text-2xl" />;
+};
+
+/* eslint-disable no-param-reassign */
+const updateReadingHistoryPost =
+  (
+    queryClient: QueryClient,
+    historyQueryKey: QueryKey,
+    update: (oldPost: ReadHistoryPost) => Partial<ReadHistoryPost>,
+  ): ((args: { id: string }) => Promise<() => void>) =>
+  async ({ id }) => {
+    const oldReadingHistory =
+      queryClient.getQueryData<ReadHistoryInfiniteData>(historyQueryKey);
+    const newItems = oldReadingHistory.pages.flatMap(({ readHistory }) => {
+      const modified = readHistory.edges.map(({ node }) => {
+        if (node.post.id === id) {
+          node.post = {
+            ...node.post,
+            ...update(node.post),
+          };
+        }
+        return { node };
+      });
+      readHistory.edges = modified;
+      return {
+        readHistory,
+      };
+    });
+    queryClient.setQueryData<ReadHistoryInfiniteData>(historyQueryKey, {
+      ...oldReadingHistory,
+      pages: newItems,
+    });
+    return () => {
+      queryClient.setQueryData<ReadHistoryInfiniteData>(
+        historyQueryKey,
+        oldReadingHistory,
+      );
+    };
+  };
+/* eslint-disable no-param-reassign */
+
+export default function PostOptionsReadingHistoryMenu({
+  post,
+  onHidden,
+  onHide,
+}: PostOptionsReadingHistoryMenuProps): ReactElement {
+  const { user } = useContext(AuthContext);
+  const queryClient = useQueryClient();
+  const historyQueryKey = ['readHistory', user?.id];
+
+  /* eslint-disable @typescript-eslint/no-unused-vars */
```

Comment:
Also here, which variable is unused?

Reply:
`isPostLinkCopied` is unused.  Is there a way to destructure only `copyPostLink` from `useCopyLink()`?

Reply:
@ayadotdev Sorry remove the `_` so it's just [, copyPostLink]